### PR TITLE
Feature/plugin input support

### PR DESCRIPTION
**Pull request by krebsni**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,7 @@ docs/api.json
 docs/changelog.md
 docs/readme.md
 git-plugins
-
+plugins/test_plugin.py
```
Comment (buehlefs) [inline .gitignore:9]: we could add e local_plugins folder to gitignore and default config of the plugin runner to allow easy development with local plugins.

File: qhana_plugin_runner/api/util.py [hunk 2]
```diff
@@ -242,3 +242,51 @@ def deserialize(
...
+            required=required,
+            **kwargs,
+        )
+        if plugin_tags:
+            self.metadata["plugin_tags"] = plugin_tags
+        if plugin_name:
+            self.metadata["plugin_name"] = plugin_name
+        if plugin_version:
+            self.metadata["plugin_version"] = plugin_version
+        self.validators[0] = FileUrlValidator(
```
Comment (buehlefs) [inline qhana_plugin_runner/api/util.py:275]: The FileUrlValidator also validates file:// urls which should not work in this case. Either adapt (and rename) the file URL validator to enable/disable file:// urls on demand or create a separate validator.

File: qhana_plugin_runner/api/util.py [hunk 3]
```diff
@@ -242,3 +242,51 @@ def deserialize(
 
 
 # TODO add plugin URL field + UI generator functions
+class PluginUrl(ma.fields.URL):
```
Comment (buehlefs) [inline qhana_plugin_runner/api/util.py:245]: Remove todo comment above if solved in this PR

File: qhana_plugin_runner/static/microfrontend.js [hunk 4]
```diff
@@ -96,6 +96,7 @@ function onDataUrlResponseMessage(data) {
  * @param {{lastHeight: number, heightUnchangedCount: number}} state 
  */
 function onPluginUrlResponseMessage(data) {
+    console.log(data);
```
Comment (buehlefs) [inline qhana_plugin_runner/static/microfrontend.js:99]: Remove debug outputs if functionality is working

File: qhana_plugin_runner/static/microfrontend.js [hunk 5]
```diff
@@ -217,6 +218,7 @@ function instrumentForm() {
         });
         form.querySelectorAll('input[data-input-type=plugin]').forEach(inputElement => {
             console.log("PLUGIN INPUT:", inputElement)
+            // TODO: request-plugin-url-info
```
Comment (buehlefs) [inline qhana_plugin_runner/static/microfrontend.js:221]: This should be relatively easy to implement in this PR

**Review by buehlefs — CHANGES_REQUESTED** [review]

**Review by PhilWun — CHANGES_REQUESTED** [review]

**Review by buehlefs — APPROVED** [review]
lgtm (aside from the unused import)

**Pull request merged**